libs/src/services/anchorAudiusData/index.ts

```diff
@@ -0,0 +1,918 @@
+import type Web3Manager from '../web3Manager'
+import { PublicKey, Keypair, SystemProgram, Transaction } from '@solana/web3.js'
+import SolanaUtils from '../solanaWeb3Manager/utils'
+
+type AnchorAudiusDataConfig = {
+  programId: string
+  adminPublicKey: string
+  adminStoragePublicKey: string
+}
+
+type OmitAndRequire<T, K extends keyof T, L extends keyof T> = Partial<
+  Omit<T, K>
+> &
+  Required<Pick<T, L>>
+
+/**
+ * AnchorAudiusData acts as the interface to solana auidus data programs from a client.
+ * It wraps methods to create transactions.
+ */
+export class AnchorAudiusData {
+  anchorAudiusDataConfig: AnchorAudiusDataConfig
+  solanaWeb3Manager: SolanaWeb3Manager
+  program: AudiusDataProgram | undefined
+  programId: PublicKey | undefined
+  adminPublicKey: PublicKey | undefined
+  adminStoragePublicKey: PublicKey | undefined
+  provider: anchor.Provider | undefined
+  web3Manager: Web3Manager
+
+  /**
+   * @param {Object} anchorAudiusDataConfig
+   *  the solana cluster RPC endpoint
+   * @param {string} anchorAudiusDataConfig.mintAddress
+   * @param {SolanaWeb3Manager} solanaWeb3Manager
+   * @param {Web3Manager} web3Manager
+   */
+  constructor(
+    anchorAudiusDataConfig: AnchorAudiusDataConfig,
+    solanaWeb3Manager: SolanaWeb3Manager,
+    web3Manager: Web3Manager
+  ) {
+    this.anchorAudiusDataConfig = anchorAudiusDataConfig
+    this.solanaWeb3Manager = solanaWeb3Manager
+    this.web3Manager = web3Manager
+  }
+
+  async init() {
+    const { programId, adminPublicKey, adminStoragePublicKey } =
+      this.anchorAudiusDataConfig
+    this.programId = SolanaUtils.newPublicKeyNullable(programId)
+    this.adminPublicKey = SolanaUtils.newPublicKeyNullable(adminPublicKey)
+    this.adminStoragePublicKey = SolanaUtils.newPublicKeyNullable(
+      adminStoragePublicKey
+    )
+    this.provider = new anchor.AnchorProvider(
+      this.solanaWeb3Manager.connection,
+      // NOTE: Method requests type wallet, but because signtransaction is not used, keypair is fine
+      Keypair.generate() as any,
+      anchor.AnchorProvider.defaultOptions()
+    )
+    this.program = new anchor.Program(
+      AudiusData.idl as Idl,
+      this.programId as anchor.web3.PublicKey,
+      this.provider
+    ) as any
+  }
+
+  // Setters
+  setAdminPublicKey(adminPublicKey: PublicKey) {
+    this.adminPublicKey = adminPublicKey
+  }
+
+  setAdminStoragePublicKey(adminStoragePublicKey: PublicKey) {
+    this.adminStoragePublicKey = adminStoragePublicKey
+  }
+
+  async sendTx(tx: Transaction) {
+    const signatures = tx.signatures
+      .filter((s) => s.signature && s.publicKey)
+      .map((s: any) => ({
+        publicKey: s.publicKey.toBase58(),
+        signature: s.signature
+      }))
+
+    const response =
+      await this.solanaWeb3Manager.transactionHandler.handleTransaction({
+        instructions: tx.instructions,
+        // TODO: Figure out
+        errorMapping: null,
```

Comment:
ooooh

Reply:
Filled this out - leveraging the anchor error parser too, so it should give us descriptive text back now